### PR TITLE
Friendship and User Associations

### PR DESCRIPTION
**Pull request by ZawadiSumba66**
In this milestone, we created associations between the user and the friendship models.  We also integrated RSpec for the user, post, comment, like, and friendship model

### CODE REVIEW

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @ZawadiSumba66 and @SamArsanios :clap: :clap:
#### Well done :muscle: The project looks great. However, some requirements are missing/not implemented correctly. Check my comments below :point_down:

#### 1) Readme:
 - Add how to clone the project to the "Getting Started" section.

#### 2) App Behaviour:
 - User should be able to invite a friend: there should be an invitation button next to the usernames in the "ALL USERS" page and likewise in each user profile page.
 - User should also be able to see the accept and reject buttons if another send a friend request to them
 - Posts timeline should include posts written by logged-in user and their friends

#### 3) Test:
 - One test is failing.
 - Add integration tests.

#### _Reviewed by Temitayo Adeleke_

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Excellent job so far, most requirements are working well. 👍 

There are still some changes that are necessary for this project to be approved.

- [ ] Your posts timeline should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [ ] One of your tests are failing, kindly add integration tests.
Screen Shot 2021-03-08 at 6 41 43 PM

- [ ] Kindly add Integration tests, great job on the unit testing, you should consider using [shoulda-matchers ](https://github.com/thoughtbot/shoulda-matchers) for your unit test, It is simple and expressive, you should try it.  Example
```
describe 'associations' do
    it { should belong_to(:category) }
  end

  describe 'validations' do
    it { should validate_presence_of(:name) }
    it { should validate_uniqueness_of(:name).scoped_to(:category_id) }
  end
```

- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  
See affected files below:
> `layouts/application.html.erb`

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 


### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    expect(page).to have_content 'Signed in successfully'
+  end
+  scenario 'when a user logs out' do
+    visit '/users/sign_in'
+    fill_in 'user_email', with: 'user4@gmail.com'
+    fill_in 'user_password', with: '123456'
+    click_on 'Log in'
+    click_on 'Sign out'
+    expect(current_path).to eql('/users/sign_in')
+  end
```
Comment (acushlakoncept) [inline spec/models/user_spec.rb]: It's not a good practice to combine unit and integration tests in one file. This should go to another file, say `spec/feature/user_spec.rb`

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### I am truly impressed with your progress, well done 👍 
Let us quickly add some finishing touches and request another review.

- [ ] You're yet to take care of the failed tests
Screen Shot 2021-03-09 at 1 32 32 PM

- [ ] Use this time to organize your specs, combining both unit and integration tests in one file is not a good practice. See the comments I have made on your codebase.

### Do this and we should be good to go.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @ZawadiSumba66**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**